pages/api/datarooms/index.ts

```diff
@@ -0,0 +1,98 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getServerSession } from "next-auth/next";
+import prisma from "@/lib/prisma";
+import { authOptions } from "../auth/[...nextauth]";
+import { CustomUser } from "@/lib/types";
+import { isUserMemberOfTeam } from "@/lib/team/helper";
+import { TeamError } from "@/lib/errorHandler";
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  if (req.method === "GET") {
+    // GET /api/datarooms
+    const session = await getServerSession(req, res, authOptions);
+    if (!session) {
+      return res.status(401).end("Unauthorized");
+    }
+    const userId = (session?.user as CustomUser).id;
+    const teamId = req.query.teamId as string;
+    //For fetching datarooms for /datarooms page
+    try {
+      //Check if user if member of team
+      await isUserMemberOfTeam({ teamId, userId });
+      const datarooms = await prisma.dataroom.findMany({
+        where: {
+          teamId,
+        },
+        include: {
+          folders: {
+            where: {
+              parentFolderId: null,
+            },
+          },
+          _count: {
+            select: {
+              files: true,
+            },
+          },
+        },
+      });
+
+      res.status(200).json({ datarooms });
+    } catch (error) {
+      if (error instanceof TeamError) {
+        return res.status(401).json({ message: "Unauthorized access" });
+      }
+      return res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
```

Comment:
this part could be replace by `errorHandler(error, res)`
which will handle both TeamError and other internal errors
check out how other endpoints implement that handler :)

Reply:
Done